pkg/apis/garden/validation/validation.go

```diff
@@ -1546,27 +1546,67 @@ func validateKubernetes(kubernetes garden.Kubernetes, fldPath *field.Path) field
 
 	return allErrs
 }
 
-func validateKubeControllerManager(kcm *garden.KubeControllerManagerConfig, fldPath *field.Path) field.ErrorList {
+func validateKubeControllerManager(kubernetesVersion string, kcm *garden.KubeControllerManagerConfig, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
+
+	k8sVersionLessThan112, err := utils.CompareVersions(kubernetesVersion, "<", "1.12")
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, kubernetesVersion, err.Error()))
+	}
 	if kcm != nil {
 		if hpa := kcm.HorizontalPodAutoscalerConfig; hpa != nil {
 			fldPath = fldPath.Child("horizontalPodAutoscaler")
-			if hpa.DownscaleDelay != nil && hpa.DownscaleDelay.Duration < 0 {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("downscaleDelay"), *hpa.DownscaleDelay, "downscale delay must not be negative"))
-			}
+
 			if hpa.SyncPeriod != nil && hpa.SyncPeriod.Duration < 1*time.Second {
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("syncPeriod"), *hpa.SyncPeriod, "syncPeriod must not be less than a second"))
 			}
 			if hpa.Tolerance != nil && *hpa.Tolerance <= 0 {
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("tolerance"), *hpa.Tolerance, "tolerance of must be greater than 0"))
 			}
-			if hpa.UpscaleDelay != nil && hpa.UpscaleDelay.Duration < 0 {
-				allErrs = append(allErrs, field.Invalid(fldPath.Child("upscaleDelay"), *hpa.UpscaleDelay, "upscale delay must not be negative"))
+			if hpa.DownscaleDelay != nil {
+				if !k8sVersionLessThan112 {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("downscaleDelay"), *hpa.DownscaleDelay, fmt.Sprintf("downscale delay is deprecated in k8s version %s", kubernetesVersion)))
+				}
+				if hpa.DownscaleDelay.Duration < 0 {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("downscaleDelay"), *hpa.DownscaleDelay, "downscale delay must not be negative"))
+				}
+			}
+			if hpa.UpscaleDelay != nil {
+				if !k8sVersionLessThan112 {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("upscaleDelay"), *hpa.UpscaleDelay, fmt.Sprintf("upscale delay is deprecated in k8s version %s", kubernetesVersion)))
+				}
+				if hpa.UpscaleDelay.Duration < 0 {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("upscaleDelay"), *hpa.UpscaleDelay, "upscale delay must not be negative"))
+				}
+			}
+			if hpa.DownscaleStabilization != nil {
+				if k8sVersionLessThan112 {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("downscaleStabilization"), *hpa.DownscaleStabilization, "downscale stabilization cannot be used in k8s versions < 1.12"))
+				}
+				if hpa.DownscaleStabilization.Duration < 1*time.Second {
+					allErrs = append(allErrs, field.Invalid(fldPath.Child("downscaleStabilization"), *hpa.DownscaleStabilization, "downScale stabilization must not be less than a second"))
```

Comment:
This validation should not be executed at all if the kubernetes version is greater or equal 1.12